.pre-commit-config.yaml

```diff
@@ -0,0 +1,21 @@
+---
+repos:
+  - repo: "https://github.com/astral-sh/ruff-pre-commit"
```

Comment:
I switched from black + pyflakes + isort to ruff, which is the current state-of-the-art for python linting and formatting. It's written in rust, opinionated in mostly sane ways, and is hella fast.